### ISSUE TITLE
calico CNI plugin can not add IP address for pod

### ISSUE THREAD

**Issue by chenww** [kubernetes/kubernetes#50156]
@kubernetes/sig-Network-bugs

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:
I have a k8s cluster that hosts 20 around services. After a couple days I observed a few services are not running, and in the following state. Delete pod and recreate pod does not help either.

> persistent-cache                        0/1       rpc error: code = 2 desc = failed to start container "abc6b44f4d76871cecb3b3300c8e70515647296a38d831662e644a184e35c511": Error response from daemon: {"message":"cannot join network of a non running container: e8c95196933ae83eeb82fc143179071d0957edd4651eca3e0b25902ad023a922"}

## kubelet logs:
```
Aug 04 16:46:06 integmaster0 kubelet[13103]: time="2017-08-04T16:46:06Z" level=info msg="Loading config from environment"
Aug 04 16:46:06 integmaster0 kubelet[13103]: 2017-08-04 16:46:06.245406 I | warning: ignoring ServerName for user-provided CA for backwards compatibility is deprecated
Aug 04 16:46:06 integmaster0 kubelet[13103]: Calico CNI checking for existing endpoint: &{{workloadEndpoint v1} {{54616} eth0 symphony.persistent-cache k8s integmaster0 dde98b2df3312651623355ec9a70eb6eb40a91352e872025d7c8b84e3dce7e57 map[calico/k8s_ns:symphony name:redis redis-sentinel:true role:master]} {[10.233.244.39/32] [] <nil> <nil> [k8s_ns.symphony] cali1dd54d1a7e0 2a:db:af:8d:75:21}}
Aug 04 16:46:06 integmaster0 kubelet[13103]: time="2017-08-04T16:46:06Z" level=info msg="Extracted identifiers for CmdAddK8s" Node=integmaster0 Orchestrator=k8s Workload=symphony.persistent-cache
Aug 04 16:46:06 integmaster0 kubelet[13103]: Calico CNI using IPs: [10.233.244.39/32]
Aug 04 16:46:06 integmaster0 kubelet[13103]: time="2017-08-04T16:46:06Z" level=info msg="Added Mac and interface name to endpoint" Workload=symphony.persistent-cache endpoint=&{{workloadEndpoint v1} {{54616} eth0 symphony.persistent-cache k8s integmaster0 dde98b2df3312651623355ec9a70eb6eb40a91352e872025d7c8b84e3dce7e57 map[calico/k8s_ns:symphony name:redis redis-sentinel:true role:master]} {[10.233.244.39/32] [] <nil> <nil> [k8s_ns.symphony] cali1dd54d1a7e0 ca:8d:e4:9d:94:28}}
Aug 04 16:46:06 integmaster0 kubelet[13103]: time="2017-08-04T16:46:06Z" level=info msg="Wrote updated endpoint to datastore" Workload=symphony.persistent-cache
Aug 04 16:46:06 integmaster0 kubelet[13103]: E0804 16:46:06.311314   13141 cni.go:294] Error adding network: cannot convert: no valid IP addresses
Aug 04 16:46:06 integmaster0 kubelet[13103]: E0804 16:46:06.311337   13141 cni.go:243] Error while adding to cni network: cannot convert: no valid IP addresses
Aug 04 16:46:06 integmaster0 kubelet[13103]: E0804 16:46:06.513709   13141 remote_runtime.go:91] RunPodSandbox from runtime service failed: rpc error: code = 2 desc = NetworkPlugin cni failed to set up pod "persistent-cache_symphony" network: cannot convert: no valid IP addresses
Aug 04 16:46:06 integmaster0 kubelet[13103]: E0804 16:46:06.513834   13141 kuberuntime_sandbox.go:54] CreatePodSandbox for pod "persistent-cache_symphony(66a5dcd9-7934-11e7-8d2e-fa163e6a8c6a)" failed: rpc error: code = 2 desc = NetworkPlugin cni failed to set up pod "persistent-cache_symphony" network: cannot convert: no valid IP addresses
Aug 04 16:46:06 integmaster0 kubelet[13103]: E0804 16:46:06.513866   13141 kuberuntime_manager.go:618] createPodSandbox for pod "persistent-cache_symphony(66a5dcd9-7934-11e7-8d2e-fa163e6a8c6a)" failed: rpc error: code = 2 desc = NetworkPlugin cni failed to set up pod "persistent-cache_symphony" network: cannot convert: no valid IP addresses
Aug 04 16:46:06 integmaster0 kubelet[13103]: E0804 16:46:06.513969   13141 pod_workers.go:182] Error syncing pod 66a5dcd9-7934-11e7-8d2e-fa163e6a8c6a ("persistent-cache_symphony(66a5dcd9-7934-11e7-8d2e-fa163e6a8c6a)"), skipping: failed to "CreatePodSandbox" for "persistent-cache_symphony(66a5dcd9-7934-11e7-8d2e-fa163e6a8c6a)" with CreatePodSandboxError: "CreatePodSandbox for pod \"persistent-cache_symphony(66a5dcd9-7934-11e7-8d2e-fa163e6a8c6a)\" failed: rpc error: code = 2 desc = NetworkPlugin cni failed to set up pod \"persistent-cache_symphony\" network: cannot convert: no valid IP addresses"
Aug 04 16:46:07 integmaster0 kubelet[13103]: I0804 16:46:07.254763   13141 kubelet.go:1912] SyncLoop (PLEG): "persistent-cache_symphony(66a5dcd9-7934-11e7-8d2e-fa163e6a8c6a)", event: &pleg.PodLifecycleEvent{ID:"66a5dcd9-7934-11e7-8d2e-fa163e6a8c6a", Type:"ContainerDied", Data:"b028ce374d9860497c26ea178789a460d7514d2f1bb3804f1c528129e5c7f40d"}
```
## my pod yml
```apiVersion: v1
kind: Pod
metadata:
  labels:
    name: redis
    redis-sentinel: "true"
    role: master
  name: persistent-cache
  namespace: symphony
spec:
  imagePullSecrets:
  - name: naas-registry-secret

  containers:
    - name: master
      image: redis:v2
      env:
        - name: MASTER
          value: "true"
      ports:
        - containerPort: 6379
      volumeMounts:
        - mountPath: /redis-master-data
          name: data
  volumes:
    - name: data
      hostPath:
        path: /var/redis-data
```
**What you expected to happen**:
pod can be created
**How to reproduce it (as minimally and precisely as possible)**:
Not sure how to reproduce this but I will give all details below for someone to help on

**Anything else we need to know?**:
I am using Calico plugin:
```
calico/cni                                                              v1.7.0              61bd0f1d1acf        3 months ago        67 MB
calico/node                                                             v1.1.3              573ddcad1ff5        3 months ago        217 MB
calico/ctl                                                              v1.1.3              4cb99a60e023        3 months ago        44 MB
```

**Environment**:
- Kubernetes version (use `kubectl version`): 1.6.4
- Cloud provider or hardware configuration**: Openstack
- OS (e.g. from /etc/os-release):
$ cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

- Kernel (e.g. `uname -a`):
Linux integmaster0 3.10.0-327.el7.x86_64 #1 SMP Thu Nov 19 22:10:57 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

- Install tools:
kubespray master branch

- Others:

**Comment by chenww** [comment]
@kubernetes/sig-Network-bugs

**Comment by caseydavenport** [comment]
@chenww I think you might be encountering this issue, which was fixed in calico/cni 1.8.0: https://github.com/projectcalico/cni-plugin/issues/349

I'd recommend upgrading to the latest Calico CNI plugin and see if you still encounter the issue.

If you do still have issues, please raise anther issue against [the Calico CNI repo](https://github.com/projectcalico/cni-plugin) and we can discuss more.

/close
